pipeline/ephys.py

```diff
@@ -133,6 +133,52 @@ class UnitPosition(dj.Part):
         """
 
 
+@schema
+class BrainAreaDepthCriteria(dj.Manual):
+    definition = """
+    -> ProbeInsertion
+    -> lab.BrainArea
+    ---
+    depth_upper: float  # (um)
+    depth_lower: float  # (um)
+    """
+
+
+@schema
+class UnitCoarseBrainLocation(dj.Computed):
+    definition = """
+    # Estimated unit position in the brain
+    -> Unit
+    ---
+    -> [nullable] experiment.BrainLocation
+    """
+
+    key_source = Unit & BrainAreaDepthCriteria
+
+    def make(self, key):
+        posy = (Unit & key).fetch1('unit_posy')
+
+        # get brain location info from this ProbeInsertion
+        brain_area, hemi, skull_ref = (experiment.BrainLocation & (ProbeInsertion.InsertionLocation & key)).fetch1(
+            'brain_area', 'hemisphere', 'skull_reference')
+
+        brain_area_rules = (BrainAreaDepthCriteria & key).fetch(as_dict=True)
+
+        coarse_brain_area = None
+        for rule in brain_area_rules:
+            if np.logical_and(posy > rule['depth_upper'], posy <= rule['depth_lower']):
+                coarse_brain_area = rule['brain_area']
+                break
```

Comment:
This doesn't detect bad rules (e.g. >1 matches). This would be user error, but it might make sense to detect, esp. since error cases would be non-deterministic depending on the fetch order of the rules.

Alternatively could validate the rules are non-overlapping before/during the loop.

 I don't think this is the only case like this in the codebase however.

Reply:
agreed, added the validation